src/Symfony/Bridge/Twig/DataCollector/TwigDataCollector.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\DataCollector;
+
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * TwigDataCollector.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class TwigDataCollector extends DataCollector implements LateDataCollectorInterface
+{
+    private $profile;
+    private $computed;
+
+    public function __construct(\Twig_Profiler_Profile $profile)
+    {
+        $this->profile = $profile;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lateCollect()
+    {
+        $this->data['profile'] = serialize($this->profile);
+    }
+
+    public function getTime()
+    {
+        return $this->getProfile()->getDuration() * 1000;
+    }
+
+    public function getTemplateCount()
+    {
+        $this->getProfile();
+
+        return $this->computed['template_count'];
+    }
+
+    public function getTemplates()
+    {
+        $this->getProfile();
+
+        return $this->computed['templates'];
```

Comment:
IMO, it's weird to call a getter before reading a property wich seem to not be related to the previous call
What's about

```
public function getComputedData($key)
{
    if (null === $this->computed ) {
        $this->computed = $this->computeData($this->getProfile());
    }

    return $this->computed[$key];
}
```